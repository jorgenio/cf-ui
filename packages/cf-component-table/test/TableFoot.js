import React from 'react';
import renderer from 'react-test-renderer';
import {
  TableFoot,
  TableFootUnstyled,
  TableRowUnstyled
} from '../../cf-component-table/src/index';
import felaTestContext from '../../../felaTestContext';
import { mount } from 'enzyme';

test('should render', () => {
  const component = renderer.create(
    felaTestContext(<TableFoot>TableFoot</TableFoot>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render extra class name', () => {
  const component = renderer.create(
    felaTestContext(<TableFoot className="extra">TableFoot</TableFoot>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should pass down CSS query props', () => {
  const wrapper = mount(
    <table>
      <TableFootUnstyled striped hover bordered condensed bare tfootIndex={0}>
        <TableRowUnstyled />
        <TableRowUnstyled />
      </TableFootUnstyled>
    </table>
  );

  expect(wrapper.childAt(0).childAt(1).prop('condensed')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('striped')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bordered')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('hover')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bare')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('tfootIndex')).toBe(0);
  expect(wrapper.childAt(0).childAt(1).prop('rowIndex')).toBe(-1);
});
