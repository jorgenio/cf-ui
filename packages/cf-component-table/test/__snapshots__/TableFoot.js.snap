// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<tfoot
  className="cf-y18mng"
>
  TableFoot
</tfoot>
`;

exports[`should render extra class name 1`] = `
<tfoot
  className="extra"
>
  TableFoot
</tfoot>
`;
